RegionalOverviews/overviews_reports_RDBES/fisheries_overviews_RDBES/scripts/parametersDefinition.R

```diff
@@ -17,22 +17,47 @@
 #
 ###################################################################
 
+# Print start message
+cat("[3]    Parameters definition")
+
+# Empty warnings from previous code
+assign("last.warning", NULL, envir = baseenv()) # Credits: https://stackoverflow.com/questions/5725106/r-how-to-clear-all-warnings
+
 ## Download data 
 params <- list(
   year = yearSelected,
   region = regionSelected, 
   logo_path = file.path("../../overviews_shiny/www/logo RCG BALTIC.PNG"), # move it to the rmd part <----------------- to do 
-  data_dir = paste0(getwd(), 'RegionalOverviews/data_RDBES/002_prepared/', downloadDataFromSPDate, '/RCG_', regionSelected),
-  CLfileName = paste0('RDBES_RCG_', regionSelected, '_CL_2021_2021_prepared_', downloadDataFromSPDate),
-  CEfileName = paste0('RDBES_RCG_', regionSelected, '_CE_2021_2021_prepared_', downloadDataFromSPDate),
+  data_dir = paste0(getwd(), '/RegionalOverviews/data_RDBES/002_prepared/', dataprepDate, "/", regionSelected, '/RCG_', regionSelected),
+  CLfileName = paste0('RDBES_RCG_', regionSelected, '_CL_2021_2021_prepared_', dataprepDate),
```

Comment:
"_some dataset names hardcoded , which should rather be changed_."
 
 Do you mean the ones at these lines ? @mrska

Reply:
If yes there is probably a work around, if the name in the data prep. is standardized we might start from there. 
I see all your other points and agree that it would be better to have the edits you mentioned. 
Btw, thanks for the review @mrska !

Reply:
Yes, I mean CLfileName and CEfileName. The part '2021_2021' should be replaced somehow, so the script works also for 2023 data for example.

Reply:
The values for the years in the hardcoded string cited above depend on the _year_start_ and _year_end_ objects in the preparation script (001_read_and_prepare_data_RDBES_CL_CE.R).

To solve this without defining these objects again, I would say that we may use the _yearSelected_ object. Two options without having tried yet:

- we act at the source, using the value in yearSelected to select the data to filter in SharePoint (this requires access and being able to list the files on the relevant web page)
- we act in the end, once the data is downloaded with downloadPreparedData we filter by year in yearSelected.

I wonder if we can't directly change the name of the data that comes out of the preparation. After doing so, we could omit the starting and ending years in the string that loads the data in parametersDefinition.R (e.g. [..] 2021_2021 [...]) and apply the second option to obtain only the data requested by the user.

In your opinion, is it essential to have the year indicated in the 001_read_and_prepare_data_RDBES_CL_CE.R output?

Any case, I need to try some code to solve, just food for thoughts for now.

Reply:
The second option sounds better to me. The only thing I'm not sure about is whether we can skip the year in the dataset name. It might work without the year indication, but I'm not 100% sure. There is a risk of some overwritings maybe? If we're preparing multiple years, we probably stroe it in just one file, but maybe not? Just my loose thoughts on it.

Reply:
Should this topic with the parameters be added to the 'to do list' or issues? Not to forget about it?

Reply:
I also prefer the second option (at least, it seems easier to me).
Having a single database hosting 1+ years in one specific location is easier to download and filter in the script that exports the report. For now I don't see any major problems.

But if I look ahead, in a few years it may become time-consuming to download a database covering many years.
At that point, option one will be better in e.g. a SharePoint scheme with multiple folders (one for each year) and a download function capable of downloading data from the right folder based on the user's selection.

It's about deciding how to structure the folder on SharePoint. Let's discuss it at the next meeting, good idea to add it to the to do list.
Alternatively, if you deem it urgent for your developments, we open an issue, circulate it via email next week and proceed based on the feedbacks.

PS. I will proceed with the code review next week. Have a nice we!